### PR TITLE
feat: switch CI and build system

### PR DESCRIPTION
**Pull request by victorolinasc**
This is the much delayed swith from travis + Cask + ert-runner to Eldev.

Sorry for the loooooooong delay on this... will try to be more present here :)

It should fix all the issues raised on #466 and some others.

Example run: https://github.com/victorolinasc/emacs-elixir/actions/runs/671131814

### CODE REVIEW

File: Eldev [hunk 1]
```diff
@@ -0,0 +1,11 @@
+; -*- mode: emacs-lisp; lexical-binding: t; no-byte-compile: t -*-
+
+;; Autodetermined by `eldev init'.
+(eldev-use-package-archive 'melpa)
+
+(eldev-use-plugin 'autoloads)
+(eldev-use-plugin 'undercover)
```
Comment (victorolinasc) [inline Eldev:7]: We might want to send statistics to coveralls or something with this. Related docs: https://github.com/doublep/eldev#undercover-plugin

File: elixir-mode.el [hunk 2]
```diff
@@ -11,7 +11,7 @@
 ;; Created: Mon Nov 7 2011
 ;; Keywords: languages elixir
 ;; Version: 2.4.0
-;; Package-Requires: ((emacs "24") (pkg-info "0.4"))
+;; Package-Requires: ((emacs "25") (pkg-info "0.6"))
```
Comment (victorolinasc) [inline elixir-mode.el:14]: As mentioned in other discussions, I've bumped from 24 to 25.

**Review by jsmestad — APPROVED** [review]
🎉

File: Eldev (hunk 1 above)
Comment (axelson) [reply Eldev:7]: Sounds good. That can probably be handled in a new PR.

**Review by axelson — APPROVED** [review]
Looks good :heart:! I retried the tests just to double-check how reliable they are.

**Pull request merged**